woocommerce-prevent-repeat-purchases.php

```diff
@@ -18,11 +18,14 @@ public function __construct() {
 		// Process the Admin Panel Saving
 		add_action( 'woocommerce_process_product_meta', [ &$this, 'write_panel_save' ] );
 		// Admin-side Scripts
-        add_action( 'admin_enqueue_scripts', [ &$this, 'admin_scripts' ] );
+		add_action( 'admin_enqueue_scripts', [ &$this, 'admin_scripts' ] );
 
 		// Don't allow the product to be purchased more than once
 		add_filter( 'woocommerce_is_purchasable', [ &$this, 'prevent_repeat_purchase' ], 10, 2 );
 
+		// Make sure the customer hasn't purchased the product before on checkout
+		add_action( 'woocommerce_before_checkout_process', [ &$this, 'before_checkout_process' ], 10, 1 );
```

Comment:
I would use the `woocommerce_check_cart_items` hook as its whats meant to check/validate cart items.